File: 0056-merge-intervals/0056-merge-intervals.py
class Solution:
    def merge(self, intervals: List[List[int]]) -> List[List[int]]:
        intervals.sort()
        res = [intervals[0]]
        for i in range(1, len(intervals)):
            prev = res[-1]
            cur = intervals[i]
            if cur[0] <= prev[1]:
                prev[1] = max(cur[1], prev[1])
            else:
                res.append(cur)
        return res
                

File: 0056-merge-intervals/0056-merge-intervals.java
class Solution {
    public int[][] merge(int[][] intervals) {
        Arrays.sort(intervals, (a, b) -> Integer.compare(a[0], b[0]));
        List<int[]> arr = new ArrayList<>();
        arr.add(intervals[0]);
        for (int i = 1; i < intervals.length; i++) {
            int[] prev = arr.get(arr.size()-1);
            int[] cur = intervals[i];
            if (cur[0] <= prev[1]) {
                prev[1] = Math.max(cur[1], prev[1]);
            } else {
                arr.add(cur);
            }
        }
        int[][] res = new int[arr.size()][];
        for (int i = 0; i < arr.size(); i++) {
            res[i] = arr.get(i);
        }
        return res;
    }
}